OpenKE Pytorch compatibility with link prediction scores

Charanyash opened the pull request.

svkeerthy reviewed (changes requested):
Can you please rename Openke directory to OpenKE? It will help in better diffing.

svkeerthy commented:
Please fix the formatting to address pre-commit check failure. You may need to install the [pre-commit package](https://pre-commit.com/) and setup the hooks given [here](https://github.com/IITH-Compilers/IR2Vec/blob/main/.pre-commit-config.yaml).

svkeerthy commented on seed_embeddings/OpenKE/LICENSE:
Please retain this license.

Charanyash replied:
Done

Charanyash replied:
Done

svkeerthy commented on seed_embeddings/OpenKE/analogies.txt:
Please retain this file.

Charanyash replied:
I did that

svkeerthy commented on seed_embeddings/OpenKE/generate_embedding_ray.py:
```diff
@@ -172,21 +206,43 @@ def findRep(src, dest, index_dir):
+        run_config=RunConfig(
+            checkpoint_config=CheckpointConfig(
+                num_to_keep=2,
+                # *Best* checkpoints are determined by these params:
```
These values are to be set right? We can check if `is_analogy` is set and select based on that.

svkeerthy commented on seed_embeddings/OpenKE/generate_embedding_ray.py:
```diff
@@ -172,21 +206,43 @@ def findRep(src, dest, index_dir):
+    # print(tuner.run_config)
     results = tuner.fit()
-
+    # print("Best config: ", results.get_best_result(metric="lose", mode="min"))
```
Results are to be written/printed right?

Charanyash replied:
Yes we can do that

svkeerthy commented on seed_embeddings/OpenKE/preprocess.py:
This file should be retained.

Charanyash replied:
Done

svkeerthy commented on seed_embeddings/OpenKE/openKE.yaml:
```diff
@@ -1,7 +1,75 @@
-name: openKE
+name: openke
```
Please change it to openKE

Charanyash replied:
Okay I will rename it

Charanyash commented:
I have updated the code with requested changes and added flags for link prediction and analogies calculations while training. Please check the latest code

svkeerthy commented on seed_embeddings/OpenKE/analogy.py:
```diff
@@ -27,6 +28,32 @@ def findVec(str1, src):
+
+    for opcode, value in entity_dict.items():
+        value = np.array(value)
+        cosineDict[opcode] = spatial.distance.cosine(vec, value)
```
Would using pairwise cosine_similarity speed up the computation? Please check. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html

svkeerthy commented on seed_embeddings/OpenKE/analogy.py:
```diff
@@ -27,6 +28,32 @@ def findVec(str1, src):
+    for opcode, value in entity_dict.items():
+        value = np.array(value)
+        cosineDict[opcode] = spatial.distance.cosine(vec, value)
+        euclDict[opcode] = spatial.distance.euclidean(vec, value)
```
Can be commented out for now.

svkeerthy commented on seed_embeddings/OpenKE/config/Trainer.py:
```diff
@@ -148,6 +197,12 @@ def run(self, link_prediction=False, test_dataloader=None, model=None, ray=True)
 
+        print("out_path : ", self.out_path)
+        if self.out_path:
+            print("Inside out_path")
```
See that these logs are cleaned up.

svkeerthy commented on seed_embeddings/OpenKE/data/TrainDataLoader.py:
```diff
@@ -126,6 +127,7 @@ def read(self):
         self.batch_t_addr = self.batch_t.__array_interface__["data"][0]
         self.batch_r_addr = self.batch_r.__array_interface__["data"][0]
         self.batch_y_addr = self.batch_y.__array_interface__["data"][0]
+        print("END OF READ")
```
Pl remove this.

svkeerthy commented on seed_embeddings/OpenKE/generate_embedding_ray.py:
```diff
@@ -66,54 +66,61 @@ def train(arg_conf):
         norm_flag=True,
     )
-
+    print("After Transe")
```
Pl remove debug logs.

svkeerthy commented on seed_embeddings/OpenKE/generate_embedding_ray.py:
```diff
@@ -206,46 +222,113 @@ def findRep(src, dest, index_dir):
-    #     )
 
-    # print("Best config: ", results.get_best_result(metric="AnalogiesScore", mode="max"))
+    if arg_conf.is_analogy:
```
Save the final embeddings corresponding to the best model.

svkeerthy reviewed (changes requested):
Please update the README
